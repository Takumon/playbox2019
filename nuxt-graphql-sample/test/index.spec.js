import { mount, shallowMount, createLocalVue } from '@vue/test-utils'
import TestTargetVue from '../pages/index.vue'

const localVue = createLocalVue()

describe('index.vue', () => {


    test('displayed repos correctly with query data', () => {
        const wrapper = shallowMount(TestTargetVue, {localVue})
        
        wrapper.setData({
            viewer: {
                repositories: {
                    nodes: {
                        name: 'test',
                        url: 'test.com',
                        id: 'foobar'
                    }
                }
            }
        })

        expect(wrapper.element).toMatchSnapshot()
    })

    test('called Apollo mutation in addStar() method', () => {
        const mutate = jest.fn().mockReturnValue({
            data: {
                value: 'testData'
            }
        });
        const wrapper = mount(TestTargetVue, {
            localVue,
            mocks: {
                $apollo: {
                    mutate,
                }
            }
        })

        wrapper.vm.addStar()
        expect(mutate).toBeCalled()
    })
})
