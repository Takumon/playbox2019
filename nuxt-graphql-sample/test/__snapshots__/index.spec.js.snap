// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue displayed repos correctly with query data 1`] = `
<section
  class="container"
>
  <div>
    <input
      name=""
      type="number"
    />
     
    <ul />
  </div>
</section>
`;
